forEach/challenges-01.test.js

```diff
@@ -0,0 +1,316 @@
+//   - The number of times the number should be added
+//   - A callback function to use to add the numbers to the array (Hint: you already defined it)
+//
+// Within the addNumbers function, invoke the callback function as many times as necessary,
+// based on the third argument of the addNumbers function.
+//
+// Return the modified array.
+// ------------------------------------------------------------------------------------------------
+
+const addValues = (arr, value) => {
+  // Solution code here...
+  arr.push(value);
+}
+
+const addNumbers = (num, arr, times, callback) => {
+  // Solution code here...
+    for(let i=0; i<times; i++){
+        callback(arr, num); 
+    }
+  return arr;
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 6
+//
+// Write a function named removeOne that takes in a number and an array. 
+// If the number divided by three has a remainder of two, pop one element off of the array.
+// Hint: you may want to look into the modulo operation.
+//
+// Then, write a function named removeElements that takes in an array and a callback.
+// This function should iterate over the array and invoke the callback once for each element in the array.
+// For this function, use a for loop, for...in syntax, or for...of syntax.
+//
+// Return the modified array.
+// ------------------------------------------------------------------------------------------------
+
+const removeOne = (num, input) => {
+  // Solution code here...s
+  if(num % 3 === 2){
+    input.pop();
+ }
+}
+
+const removeElements = (input, callback) => {
+  // Solution code here...
+  for(let i of input) {
+    callback(i, input);
+  }
+  return input;
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 7
+//
+// Write a function named removeWithForEach that produces the same output as challenge 6, but uses forEach.
+// ------------------------------------------------------------------------------------------------
+
+const removeWithForEach = (input, callback) => {
+  // Solution code here...
+  input.forEach((item) => {
+    callback(item, input);
+  });
+  return input;
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 8
+//
+// Write a function named removeWithAnon that produces the same output as challenges 6 and 7.
+//
+// This function should use forEach again, but rather than taking in a callback as an argument,
+// define an anonymous function as the argument to forEach.
+//
+// This anonymous function should accept up to three arguments: the element, the index, and the array.
+// ------------------------------------------------------------------------------------------------
+
+const removeWithAnon = (input) => {
+  // Solution code here...
+  input.forEach((ele, index, arr) => {
+    if(ele % 3 === 2){
```

Comment:
if you are not going to use the other parameters, they are unnecessary and don't need to be included